src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -903,6 +903,27 @@ impl ScalarFunc {
             Ok(Some(Cow::Borrowed(b"")))
         }
     }
+
+    pub fn instr<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<i64>> {
+        let arg0: Cow<'a, [u8]> = try_opt!(self.children[0].eval_string(ctx, row));
+        let s: &str = box_try!(::std::str::from_utf8(&arg0));
+        if s.is_empty() {
+            return Ok(Some(-1));
+        }
+        let arg1: Cow<'a, [u8]> = try_opt!(self.children[1].eval_string(ctx, row));
+        let res: &str = box_try!(::std::str::from_utf8(&arg1));
+        if res.is_empty() {
+            return Ok(Some(-1));
+        }
+        match s.find(res) {
+            Some(x) => Ok(Some(x as i64)),
+            None => Ok(Some(-1)),
```

Comment:
According to https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_instr, if it does not find, it should return `0`.

Reply:
> It seems this PR and #4316 do the same thing?

Yes，They do the same thing. This is my first time to PR. And I have some problems in sigh-off & CI。does all the commit need verified tag? How can I do when I have already committed without -s before。And I use rust-toolchain nighty-2019-03-04 to make build。I can pass the test in local. Should  I commit all the change to tikv (include all the file)？ I just commit build_string.rs & scalar_function.rs。I'll change the code you have told. Thanks for your replay.

Reply:
@shaddockzy I am not sure the following answers can solve your problems, feel free to ask help if you have any problem. 
> I have some problems in sigh-off

You can try `git commit --amend -s`.
> Should I commit all the change to tikv

You only need to commit the changes related to this PR.
> I can pass the test in local

CI complains about the unreachable pattern `ScalarFuncSig::Instr`, you can run `make clippy` to see it in your local environment. 
> I use rust-toolchain nighty-2019-03-04

You don't need to worry about the rust-toolchain. We have a file named `rust-toolchain` to help you use the right toolchain.
> does all the commit need verified tag?

I think the answer is no.

Reply:
Thanks for your replay. I have changed the result of the func & change the test func.  I have tried some method from google and I try the **git commit --amend -s** command and pull it again to the remote. But there still have DCO problem in it. I even can see the sign-off message in my GitHub-desktop app. Unfortunately. I can't run the "make Clippy" even I have already installed the Clippy compoment。The func is easy but the PR is hard.....  emmmmmm.

Reply:
What kind of error does the clippy throw? I think before you send a commit, you should run `cargo fmt` first.

Reply:
> What kind of error does the clippy throw? I think before you send a commit, you should run `cargo fmt` first.

I have already run the clippy and cargofmt，I can run it in local。But in CI there still show the merge conflict in it.  and I even can see the sign-off message in this web，but DCO is still not pass。I just use the commit -s -m。I didn't find the reason about this.

Reply:
> What kind of error does the clippy throw? I think before you send a commit, you should run `cargo fmt` first.

![e8f0a08b-514c-4900-b1f3-e52468bf2881](https://user-images.githubusercontent.com/16465481/53975268-6da84480-413f-11e9-88fd-85b29cb01d23.png)
When i use git commit --amend -s . I have this.I think I already have the sigh-off.

Reply:
@shaddockzy CI failed because a test failed now. And I find these are some commits which sign-off are missing. e.g. add instr function (73d4b62).